lib/Mojo/WebSocketProxy/Dispatcher.pm

```diff
@@ -83,79 +82,66 @@ sub on_message {
     my $req_storage = {};
     $req_storage->{args} = $args;
 
-    my $result = Mojo::WebSocketProxy::Parser::parse_req($c, $req_storage);
+    # We still want to run any hooks even for invalid requests.
+    if(my $err = Mojo::WebSocketProxy::Parser::parse_req($c, $req_storage)) {
+        $c->send({json => $err}, $req_storage);
+        return $c->_run_hooks($config->{after_dispatch} || [])->retain;
+    }
+    my $result;
```

Comment:
This appears to be unused, and shadows a `my $result` later in the `->then` blocks

Reply:
Done 73bbca4